graphbolt/src/csc_sampling_graph.cc

```diff
@@ -214,5 +223,26 @@ torch::Tensor Pick(
   return picked_neighbors;
 }
 
+torch::Tensor PickEtype(
+    int64_t offset, int64_t num_neighbors, const std::vector<int64_t>& fanouts,
+    bool replace, const torch::TensorOptions& options,
+    const torch::Tensor& type_per_edge) {
+  std::vector<torch::Tensor> picked_neighbors_per_etype(
+      fanouts.size(), torch::tensor({}, options));
+  for (int64_t etype_end = offset, etype_begin = offset;
```

Comment:
This is a very abnormal for loop.
Do we have to do this?

Reply:
Changed, please take a look again